util/http/retry.go

```diff
@@ -83,7 +84,7 @@ func RetryStrategy(ctx context.Context, resp *http.Response, err error) (bool, e
 	retry, err2 := rhttp.DefaultRetryPolicy(ctx, resp, err)
 	if retry {
 		log.G(ctx).WithFields(logrus.Fields{
-			"error":    err,
+			"error":    logutil.RedactHTTPQueryValues(err),
```

Comment:
This feels error prone (what if we add a new log somewhere else?)

Similar to my other comment, I think we should do this in a structured way and I think it should be embedded in our http.Client. We might need to defer the embedding it until after #759.

Reply:
Makes sense. I can create a task on the tracking issue for having this within the unified client.

Just for completeness, another approach considered was to utilize `logrus.Hook`s to have the redaction run on each log message. This might alleviate the concern for any added logs. The trade-off is taking a direct dependency on `logrus` (We already do this in other parts of the snapshotter). LMK if you have thoughts on that approach or if the current approach is fine.

Reply:
Some of these errors escape over grpc. I think we should redact as close to the source as possible. In fact, we should probably be redacting `err2` before we return it at the end of this function.

Reply:
`err2` should only be context deadline exceeded or context cancelled error, but overhead is minimal so no harm in checking either.

Reply:
Just realized that we would still be implicitly logging the error if we fail on the last attempt: https://github.com/hashicorp/go-retryablehttp/blob/main/client.go#L759-L768. 

We log the error on every retry. If we have expired our retries, the rhttp client will just return the error and that will end up bubbling up to the call site. The `Client` has an [`ErrorHandler`](https://github.com/hashicorp/go-retryablehttp/blob/main/client.go#L422) attached, that will be called when we have exhausted our retries. We should set create an `ErrorHandler` that will redact parts of the error before returning.

Reply:
@turan18, good catch. I'll look into covering that one as well with this PR.

Reply:
@turan18, added error handler to latest commit. PTAL.

Reply:
LGTM thanks!